check/pytest-changed-files

```diff
@@ -61,11 +61,16 @@ changed=$(git diff --name-only ${rev} -- \
     | sort \
     | uniq \
 )
-num_changed=$(echo -e "${changed}" | wc -w)
+if git diff --name-only "${rev}" -- | grep "__init__\.py$" > /dev/null; then
+  # Include global API tests when an __init__ file is touched.
+  changed+=('docs/docs_coverage_test.py')
+  changed+=('cirq/protocols/json_test.py')
+fi
+num_changed=$(echo -e "${changed[@]}" | wc -w)
 
 # Run it.
-echo "Found ${num_changed} differing files with associated tests." >&2
-if [ -z "${changed}" ]; then
+echo "Found ${num_changed} test files associated with changes." >&2
+if [ "${num_changed}" -eq 0 ]; then
     exit 0
 fi
-pytest ${rest} ${changed}
+pytest ${rest} ${changed[@]}
```

Comment:
I don't think `[@]` has any effect unless you place the variable in double quotes: `"${changed[@]}"`.

Reply:
reverted

Reply:
Restored. Broke tests.